### PR TITLE
Added schedulingPolicy api.

### PR DESCRIPTION
**Pull request by k82cn**
Signed-off-by: Da K. Ma <klaus1982.cn@gmail.com>

part of #11

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/common/13)

### CODE REVIEW

**Comment by k82cn** [comment]
Add api firstly, will add implementation later.

**Comment by k82cn** [comment]
> Could we support advanced use cases? For example, we want to run PS in different machines and place workers around PS.

sure, we have implemented this internally; will publish it later.

**Comment by johnugeorge** [comment]
You also discussed about adding queue name.  Is it planned?

**Comment by k82cn** [comment]
yes, queue is a key feature in the pipeline. I will add integration when it is stable, e.g. `QueueName`.

File: operator/v1/types.go [hunk 1]
```diff
@@ -179,7 +179,13 @@ type RunPolicy struct {
...
-        // TODO - Implement Scheduling policy.
-	// See https://github.com/kubeflow/tf-operator/issues/916#issuecomment-458729706
-        // SchedulingPolicy *SchedulingPolicy `json:"schedulingPolicy,omitempty"`
+	// SchedulingPolicy defines the policy related to scheduling, e.g. gang-scheduling
+	// +optional
+	SchedulingPolicy *SchedulingPolicy `json:"schedulingPolicy,omitempty"`
+}
+
+// SchedulingPolicy encapsulates various scheduling policies of the distributed training
+// job, for example `minAvailalbe` for gang-scheduling.
```
Comment (richardsliu) [inline operator/v1/types.go]: Nit: typo here for minAvailable

**Comment by gaocegege** [comment]
@k82cn Could you please fix the typo, then I think we could merge it. Thanks

**Comment by k82cn** [comment]
fixed typo :)

**Comment by k82cn** [comment]
Sorry for post update: there's a "feature" in upstream, named `SchedulingPolicy`, which is used to define whether a scheduler feature can be enabled, e.g. toleration. Do we need to have another term for that, e.g. `SchedulingSpec`?

xref https://github.com/kubernetes/community/pull/1937

**Comment by gaocegege** [comment]
@k82cn Thanks for the information, Kubeflow requires k8s 1.8, does this feature exist in 1.8?

**Comment by k82cn** [comment]
> Thanks for the information, Kubeflow requires k8s 1.8, does this feature exist in 1.8?

nop, but this may make other contributor confuse.

BTW, I think it should be client-go 1.8, k8s 1.11 :)

**Pull request merged**